src/openrct2/peep/Guest.cpp

```diff
@@ -5103,6 +5103,55 @@ void Guest::UpdateRide()
 static void peep_update_walking_break_scenery(Peep* peep);
 static bool peep_find_ride_to_look_at(Peep* peep, uint8_t edge, uint8_t* rideToView, uint8_t* rideSeatToView);
 
+/**
+ *
+ *  rct2: 0x699F5A
+ * al:thoughtType
+ * ah:thoughtArguments
+ * esi: peep
+ */
+void Peep::InsertNewThought(PeepThoughtType thoughtType, uint8_t thoughtArguments)
+{
+    PeepActionType action = PeepThoughtToActionMap[thoughtType].action;
```

Comment:
Because Peep already contains a field called `action` this is causing issues. Rename this variable to `newAction` and it should fix the compilation